### ISSUE TITLE
Add DIR_O to makefile

### ISSUE THREAD

**Issue by ArkadiuszMichalski** [ScintillaOrg/lexilla#50]
Would be nice to have `DIR_O` in [makefile](https://github.com/ScintillaOrg/lexilla/blob/master/src/makefile), same as we see in [`lexilla.mak`](https://github.com/ScintillaOrg/lexilla/blob/master/src/lexilla.mak). Without it, I can't generate files by various tools without calling `clean` first. And this means that when I go back to the previous tool, I have to do the whole generation again.

@nyamatongwe Any objection to also change `include deps.mak` to native solution by using `-MMD` option for `$(CXX)`? Not everyone has Python installed to update a permanent `deps.mak` file. Alternatively this behaviour can be forced with some variable passed on the command line.

**Comment by nyamatongwe** [comment]
The patch contains many unnecessary changes to spacing and indentation which makes it more difficult to review.

> Any objection to also change include deps.mak to native solution by using -MMD option for $(CXX)?

`-MM` was used in the past and it was major pain point, particularly with Win32/Unix cross-platform issues, which is why the Python scripts were developed. There is no way I'm going back to that even as an option.

> Not everyone has Python installed

Python is required for Scintilla and Lexilla development when making changes to the code.

**Comment by ArkadiuszMichalski** [comment]
I make changes for spacing to be more consistent, because now there's a lottery with it. But if the current state is to be expected, I will not change it.

>There is no way I'm going back to that even as an option.

I added the DEPS option, but I can remove it, the whole thing will simplify a bit. I will create a new PR if adding `DIR_O` is still accepted.

**Comment by rdipardo** [comment]
@ArkadiuszMichalski,

> I will create a new PR . . . 

Looks like a lot of work! Here's a script that might save some time:

```sh
#!/bin/sh
set -ex

git remote add upstream https://github.com/ScintillaOrg/lexilla
git fetch upstream --no-tags
git reset --hard upstream/master
git checkout upstream/master -- .

#~ fetch and apply all the non-white-space changes from PR #51 
curl --silent https://patch-diff.githubusercontent.com/raw/ScintillaOrg/lexilla/pull/51.diff \ |
  git apply --ignore-whitespace -v -

```

**Comment by rdipardo** [comment]
My mistake: it's `git diff` that actually removes the spaces changes; `git apply` can only be told to ignore *conflicts* in spaces changes. (See [the edited script](https://github.com/ScintillaOrg/lexilla/issues/50#issuecomment-1004557958))

**Comment by ArkadiuszMichalski** [comment]
@rdipardo Thx for advice.

@nyamatongwe New PR has a minimum amount of changes to make `DIR_O` work.

**Comment by nyamatongwe** [comment]
Committed feature.

**Comment by nyamatongwe** [comment]
Included in 5.1.5 release.
